meilisearch-http/src/routes/setting.rs

```diff
@@ -1,107 +1,569 @@
+    pub index_new_fields: Option<bool>,
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "lowercase")]
-pub enum RankingOrdering {
-    Asc,
-    Dsc,
+pub async fn update_all(mut ctx: Request<Data>) -> SResult<Response> {
+    ctx.is_allowed(SettingsWrite)?;
+    let index = ctx.index()?;
+    let settings_update: UpdateSettings =
+        ctx.body_json().await.map_err(ResponseError::bad_request)?;
+    let db = &ctx.state().db;
+
+    let settings = Settings {
+        ranking_rules: Some(settings_update.ranking_rules),
+        ranking_distinct: Some(settings_update.ranking_distinct),
+        attribute_identifier: Some(settings_update.attribute_identifier),
+        attributes_searchable: Some(settings_update.attributes_searchable),
+        attributes_displayed: Some(settings_update.attributes_displayed),
+        stop_words: Some(settings_update.stop_words),
+        synonyms: Some(settings_update.synonyms),
+        index_new_fields: Some(settings_update.index_new_fields),
+    };
+
+    let mut writer = db.update_write_txn()?;
+    let update_id = index.settings_update(&mut writer, settings.into())?;
+    writer.commit()?;
+
+    let response_body = IndexUpdateResponse { update_id };
+    Ok(tide::Response::new(202).body_json(&response_body).unwrap())
 }
 
-pub type RankingOrder = Vec<String>;
-pub type DistinctField = String;
-pub type RankingRules = HashMap<String, RankingOrdering>;
+pub async fn delete_all(ctx: Request<Data>) -> SResult<Response> {
+    ctx.is_allowed(SettingsWrite)?;
+    let index = ctx.index()?;
+    let db = &ctx.state().db;
+    let mut writer = db.update_write_txn()?;
+
+    let settings = SettingsUpdate {
+        ranking_rules: UpdateState::Clear,
+        ranking_distinct: UpdateState::Clear,
+        attribute_identifier: UpdateState::Clear,
+        attributes_searchable: UpdateState::Clear,
+        attributes_displayed: UpdateState::Clear,
+        stop_words: UpdateState::Clear,
+        synonyms: UpdateState::Clear,
+        index_new_fields: UpdateState::Clear,
+    };
+
+    let update_id = index.settings_update(&mut writer, settings)?;
+
+    writer.commit()?;
 
-pub async fn get(ctx: Context<Data>) -> SResult<Response> {
+    let response_body = IndexUpdateResponse { update_id };
+    Ok(tide::Response::new(202).body_json(&response_body).unwrap())
+}
+
+#[derive(Default, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct GetRankingSettings {
+    pub ranking_rules: Option<Vec<String>>,
+    pub ranking_distinct: Option<String>,
+}
+
+pub async fn get_ranking(ctx: Request<Data>) -> SResult<Response> {
     ctx.is_allowed(SettingsRead)?;
     let index = ctx.index()?;
+    let db = &ctx.state().db;
+    let reader = db.main_read_txn()?;
 
+    let ranking_rules = match index.main.ranking_rules(&reader)? {
+        Some(rules) => Some(rules.iter().map(|r| r.to_string()).collect()),
```

Comment:
Couldn't we replace this `map` by a `cloned` ?

Reply:
apparently `clone` did not work